lib/pluck_each.rb

```diff
@@ -29,12 +32,21 @@ def pluck_in_batches(*column_names)
 
       loop do
         batch = batch_relation.pluck(*column_names)
-        ids = batch.map(&:first)
+        ids = batch.map { |record| record.at(id_position_in_response) }
 
         break if ids.empty?
 
         primary_key_offset = ids.last
 
+        unless id_in_columns_requested
+          batch.collect! do |record|
+            record.delete_at(id_position_in_response)
+            record
+          end
+        end
+
+        batch.flatten! if string_column_names.size == 1
```

Comment:
I think this is only necessary when `id` was not requested. Perhaps it should be inside the `unless` block?